src/lib/codap-helper.ts

```diff
@@ -42,36 +46,13 @@ export class CodapHelper {
     return [];
   }
 
-  // if passed "foo" returns "foo-1"
-  // if passed "foo-bar-23" returns "foo-bar-24"
-  static incrementName(name: string) {
-    if (/-(\d*)$/.test(name)) {
-      const count = /-(\d*)$/.exec(name)![1];
-      const next = "" + (parseInt(count, 10) + 1);
-      return name.replace(/\d*$/, next);
-    } else {
-      return name + "-1";
-    }
-  }
-
-  static async createUniqueDataContext(dataContextPrefix: string, title: string) {
-    const contexts = await this.getDataContextList();
-    const contextNames: string[] = contexts.map((c: DataContext) => c.name);
-    let newDataContextName = dataContextPrefix;
-    while (contextNames.indexOf(newDataContextName) > -1) {
-      newDataContextName = this.incrementName(newDataContextName);
-    }
-    const newContext = await this.createDataContext({ name: newDataContextName, title });
-    return newContext;
-  }
-
-  static async createDataContext(dataContextSpec: DataContext): Promise<DataContext | null> {
-    const { name, title, collections } = dataContextSpec;
+  static async createDataContext(dataContextSpec: DataContextCreation): Promise<DataContext | null> {
+    const { title, collections } = dataContextSpec;
     const res = await codapInterface.sendRequest({
       action: "create",
       resource: "dataContext",
       values: {
-        name,
+        name: randomize("a0", 10),
```

Comment:
It may not matter, but internally CODAP is using a library called nanoid() to generate base-64 style IDs with 2^6 bits per character. Unlike the 6-character code, this one isn't user-visible.

Reply:
Are you suggesting using that library? We do need to pass in something for the name--it would be nice if CODAP allowed us to omit it, but that's not currently possible.